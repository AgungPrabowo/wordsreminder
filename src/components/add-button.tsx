import React from 'react'
import { StyleSheet, TouchableHighlight } from 'react-native'
import { PlusIcon } from '@components/svg/plus-icon'

type Props = {
  onPress: () => void
  testID?: string
  children?: never
}

const AddButton = ({ onPress, testID }: Props) => (
  <TouchableHighlight
    style={styles.addButton}
    onPress={onPress}
    underlayColor="transparent"
    testID={testID}
  >
    <PlusIcon />
  </TouchableHighlight>
)

type Style = {
  addButton: import('react-native').ViewStyle
}

const styles = StyleSheet.create<Style>({
  addButton: {
    position: 'absolute',
    bottom: 20,
    right: 20,
  },
})

export { AddButton }
