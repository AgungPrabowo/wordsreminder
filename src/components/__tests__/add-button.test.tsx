import React from 'react'
import { TouchableHighlight } from 'react-native'
import { render, fireEvent, shallow } from 'react-native-testing-library'
import { AddButton } from '../add-button'

describe('AddButton', () => {
  it('should render', () => {
    const onPress = jest.fn()
    const { getByType } = render(<AddButton onPress={onPress} />)

    const button = getByType(TouchableHighlight)
    fireEvent.press(button)

    expect(onPress).toHaveBeenCalledTimes(1)
  })

  it('should match snapshot', () => {
    const onPress = () => undefined
    const { output } = shallow(<AddButton onPress={onPress} />)
    expect(output).toMatchSnapshot()
  })
})
