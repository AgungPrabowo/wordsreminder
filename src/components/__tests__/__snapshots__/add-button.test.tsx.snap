// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton should match snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  onPress={[Function]}
  style={
    Object {
      "bottom": 20,
      "position": "absolute",
      "right": 20,
    }
  }
  underlayColor="transparent"
>
  <PlusIcon />
</TouchableHighlight>
`;
